### PR TITLE
feat(staking): add earnings projections and steps to the staking and unstaking modal

### LINKED ISSUE #9258: Staking: Returns projection
### User Story

As an Ethereum user
I want to stake Ethereum
So that I can earn a return on my hodlings

### Acceptance Criteria

Background: Yearly rate is dynamic

Given I am on the 'Stake Ethereum' modal
And the Yearly rate is 5%
When I enter '1' into the ETH column
Then I should see an estimated year gain of '0.05 ETH = $82.27*'

### [Screens](https://www.figma.com/file/hZ22ckyTRlCYRLTNUbTdWO/Exploring-Accounts?type=design&node-id=6218-23908&mode=design&t=2gxlGVuOcreATUC3-4)
image

### LINKED ISSUE #9259: Staking: Staking pool waiting period
### User Story

As an Ethereum user
I want to stake Ethereum
So that I can earn a return on my hodlings

### Acceptance Criteria

Background: 'Add to staking pool' value is dynamic

Given I am on the 'Stake Ethereum' modal
Then I should see '35 days' next to 'Add to staking pool'

### [Screens](https://www.figma.com/file/hZ22ckyTRlCYRLTNUbTdWO/Exploring-Accounts?type=design&node-id=6218-23908&mode=design&t=2gxlGVuOcreATUC3-4)
image

### LINKED ISSUE #13939: Staking: Earnings projections + steps
# Problem
1. Users are unaware of what can be expected as an earning projection with their staked Ethereum
2. Users are unclear where they stand in the staking process and how long it takes

# Solution

Compound interest tooltip: Your rewards are automatically restaked.

Image

Image

### LINKED ISSUE #13940: Staking: Unstaking steps in modal
# Problem
Users miss that unstaking has multiple steps (i.e. pay multiple network fees)

# Solution

Image

### PR DESCRIPTION
**Pull request by dev-pvl**
## Description

- Combined two issues in one PR as they share the same components.
- Added earnings projection with compound interest link to clarify expected rewards in the staking modal.
- Updated unstaking steps in the modal.
- Refactored StakeEthInANutshellModal for cleaner structure and maintainability.

## Related Issue
resolve https://github.com/trezor/trezor-suite/issues/13939
resolve https://github.com/trezor/trezor-suite/issues/13940
resolve #9258 (outdated design)
resolve #9259 (outdated design)

## Screenshots:
Screenshot
Screenshot2
Screenshot3
Screenshot4
Screenshot5
Screenshot6
Screenshot7

### CODE REVIEW

File: packages/suite/src/components/suite/StakingProcess/InfoCard.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const ContentWrapper = styled.div`
+    padding-top: ${spacingsPx.lg};
+`;
+
+interface InfoCardProps {
+    heading: React.ReactNode;
+    content: React.ReactNode;
+}
+
+export const InfoCard = ({ heading, content }: InfoCardProps) => {
```
Comment (seibei-iguchi) [inline packages/suite/src/components/suite/StakingProcess/InfoCard.tsx]: Please use CollapsibleBox for this. See stakeethinnushellmodal (or what's the name:)

**Review by seibei-iguchi — CHANGES_REQUESTED** [review]

**Comment by seibei-iguchi** [comment]
I think you used an old version of develop for the fork. Please rebase and use the changes we did in the staking modal (using collapsiblebox and list components and so on)

File: packages/suite/src/components/suite/StakingProcess/InfoCard.tsx (hunk 1 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/StakingProcess/InfoCard.tsx]: done https://github.com/trezor/trezor-suite/pull/14713/commits/43c8cfd532d265b225b284aa487560b4b30adf04

**Comment by tomasklim** [comment]
The default state now of both `Estimated gains` and `Once you confirm` are closed. 
Screenshot 2024-10-08 at 19 00 58

Do we want the first box to be opened by default? @jirih-stsh 

Btw there is a problem that before you fill the amount, gains are 0 which is stupid. But when it is closed, it looks also stupid. 
So options:
1. Hide zeros
2. Maybe let's move `Once you confirm` to the first place and have it open by default?
3. Other?
Screenshot 2024-10-08 at 18 59 58

**Review by tomasklim — COMMENTED** [review]
@dev-pvl please introduce max amount to stake. It is breaking UI. Is there a limit by contract? If not, let's set 1M? 😀 
Screenshot 2024-10-08 at 19 03 35

File: packages/suite/src/support/messages.ts [hunk 2]
```diff
@@ -7312,6 +7329,26 @@ export default defineMessages({
...
+        id: 'TR_STAKING_YOU_ARE_HERE',
+        defaultMessage: "You're here",
+    },
+    TR_STAKING_CONSOLIDATING_FUNDS: {
+        id: 'TR_STAKING_CONSOLIDATING_FUNDS',
+        defaultMessage: 'Consolidating your {symbol} for you',
+    },
+    TR_STAKING_YOUR_UNSTAKED_FUNDS: {
+        id: 'TR_STAKING_YOUR_UNSTAKED_FUNDS',
+        defaultMessage: 'Your unstaked {symbol} is ready for  you',
```
Comment (tomasklim) [inline packages/suite/src/support/messages.ts]:
```suggestion
        defaultMessage: 'Your unstaked {symbol} is ready for you',
```

File: packages/suite/src/components/suite/StakingProcess/InfoRow.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import { Badge, Column, Paragraph, Row, useElevation } from '@trezor/components';
+import { Elevation, mapElevationToBorder, spacings, spacingsPx } from '@trezor/theme';
+import styled from 'styled-components';
+import { Subheading } from './Subheading';
+import { RowContent, RowSubheading } from './types';
+
+const ListItem = styled.li<{
```
Comment (seibei-iguchi) [inline packages/suite/src/components/suite/StakingProcess/InfoRow.tsx]: This component is not needed, just use `List.Item`

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx [hunk 4]
```diff
@@ -2,9 +2,22 @@ import styled from 'styled-components';
...
+    H3,
+    List,
+    Row,
+    useMediaQuery,
+    variables,
+} from '@trezor/components';
+import { spacings } from '@trezor/theme';
+import { UnstakingInfo } from 'src/components/suite/StakingProcess/UnstakingInfo';
 
 const StyledModal = styled(Modal)`
```
Comment (seibei-iguchi) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: Please use NewModal from `components` like in the StakeModal.

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx [hunk 5]
```diff
@@ -23,12 +26,15 @@ export const StakeModal = ({ onCancel }: StakeModalModalProps) => {
     return (
         <StakeEthFormContext.Provider value={stakeEthContextValues}>
             <NewModal
-                size="tiny"
+                size="large"
                 heading={<Translation id="TR_STAKE_ETH" />}
                 onCancel={onCancel}
                 bottomContent={<StakeButton />}
```
Comment (seibei-iguchi) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx:32]: Please remove the isFullWidth prop from the button.

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+import styled from 'styled-components';
+import { spacings, spacingsPx } from '@trezor/theme';
+import { calculateGains } from 'src/utils/suite/stake';
+import { HELP_CENTER_ETH_STAKING } from '@trezor/urls';
+
+const Heading = styled.div`
+    margin-bottom: ${spacingsPx.xl};
+`;
+
+const FormattedCryptoAmountWrapper = styled.div`
```
Comment (seibei-iguchi) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx]: Use Text component instead.

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    color: ${({ theme }) => theme.textPrimaryDefault};
+`;
+
+const FiatValueWrapper = styled.div`
```
Comment (seibei-iguchi) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx]: Use Paragraph component instead.

**Review by seibei-iguchi — DISMISSED** [review]

**Comment by tomasklim** [comment]
@dev-pvl so the decision is made. In staking window, please move `Once you confirm` to the first position and make it open by default 🙏 Thanks.

Zeros in this case can stay there as they are

File: packages/suite/src/components/suite/StakingProcess/InfoRow.tsx (hunk 3 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/StakingProcess/InfoRow.tsx]: removed, https://github.com/trezor/trezor-suite/pull/14713/commits/aa21e6e49b2370a3e9bbb5035db474d51a1e8519

File: packages/suite/src/support/messages.ts (hunk 2 above)
Comment (dev-pvl) [reply packages/suite/src/support/messages.ts]: fixed

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx (hunk 4 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: added NewModal to unstake, https://github.com/trezor/trezor-suite/pull/14713/commits/aa21e6e49b2370a3e9bbb5035db474d51a1e8519

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx (hunk 5 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakeModal.tsx:32]: removed from staking and unstaking modals, https://github.com/trezor/trezor-suite/pull/14713/commits/aa21e6e49b2370a3e9bbb5035db474d51a1e8519

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx (hunk 6 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx]: https://github.com/trezor/trezor-suite/pull/14713/commits/aa21e6e49b2370a3e9bbb5035db474d51a1e8519

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx (hunk 7 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx]: https://github.com/trezor/trezor-suite/pull/14713/commits/aa21e6e49b2370a3e9bbb5035db474d51a1e8519

**Comment by dev-pvl** [comment]
@tomasklim added max amount validation and changed the order of cards. Also, I made the first card open by default as was agreed. https://github.com/trezor/trezor-suite/pull/14713/commits/aa21e6e49b2370a3e9bbb5035db474d51a1e8519

File: packages/suite/src/constants/suite/ethStaking.ts [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import { BigNumber } from '@trezor/utils/src/bigNumber';
 
 export const MIN_ETH_AMOUNT_FOR_STAKING = new BigNumber(0.1);
+export const MAX_ETH_AMOUNT_FOR_STAKING = new BigNumber(1000000); // 1M
```
Comment (tomasklim) [inline packages/suite/src/constants/suite/ethStaking.ts]: 1_000_000 and then no comment is needed

**Review by tomasklim — CHANGES_REQUESTED** [review]
There are divs in paragraphs => errors in consoles
Screenshot 2024-10-11 at 8 57 55
Screenshot 2024-10-11 at 8 57 50

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx (hunk 4 above)
Comment (tomasklim) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: Please move unstake button to the footer of the modal, same as stake button

File: packages/suite/src/utils/suite/validation.ts [hunk 9]
```diff
@@ -149,6 +150,15 @@ export const validateMin =
         }
     };
 
+export const validateMax =
```
Comment (tomasklim) [inline packages/suite/src/utils/suite/validation.ts]: this is too staking specific. How about validateStakingMax and moving it to some staking eth utils?

File: packages/suite/src/utils/suite/stake.ts [hunk 10]
```diff
@@ -619,7 +619,7 @@ export const getChangedInternalTx = (
     return internalTransfer ?? null;
 };
 
-export const calculateGains = (input: string, apy: number, divisor: number) => {
+export const calculateGains = (input: string | BigNumber, apy: number, divisor: number) => {
```
Comment (tomasklim) [inline packages/suite/src/utils/suite/stake.ts]: not needed change

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx [hunk 11]
```diff
@@ -35,16 +30,16 @@ export const UnstakeModal = ({ onCancel }: UnstakeModalModalProps) => {
     if (!account || status !== 'loaded') return null;
 
     return (
-        <StyledModal
-            isCancelable
+        <NewModal
```
Comment (tomasklim) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: Move unstake button to the footer

**Review by tomasklim — CHANGES_REQUESTED** [review]
Before new changes, please rebase to current develop and squash. Thank you 🙏

**Comment by tomasklim** [comment]
please remove the gap under fee 
Screenshot 2024-10-11 at 9 25 57

Could we provide gains calculation even when user does not have enough balance?
Screenshot 2024-10-11 at 9 25 30

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx (hunk 11 above)
Comment (tomasklim) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: Screenshot 2024-10-11 at 9 28 28

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx [hunk 12]
```diff
@@ -15,18 +23,43 @@ interface UnstakeModalModalProps {
...
         >
-            <UnstakeModalContent selectedAccount={selectedAccount} />
-        </StyledModal>
+            <Grid columns={isBelowTablet ? 1 : 2} gap={spacings.lg}>
+                <UnstakeModalContent selectedAccount={selectedAccount} />
+                <Column gap={spacings.sm} alignItems="normal">
+                    <Card paddingType="small">
+                        <CollapsibleBox
+                            heading={
+                                <Row gap={spacings.xs}>
```
Comment (seibei-iguchi) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: Not needed, I think?

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx (hunk 4 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: created a separate component for the unstake button, https://github.com/trezor/trezor-suite/pull/14713/commits/03dbbc780d1e2c8bcce13f33e84bcfa328df7e64

File: packages/suite/src/constants/suite/ethStaking.ts (hunk 8 above)
Comment (dev-pvl) [reply packages/suite/src/constants/suite/ethStaking.ts]: you're right it looks better https://github.com/trezor/trezor-suite/pull/14713/commits/03dbbc780d1e2c8bcce13f33e84bcfa328df7e64

File: packages/suite/src/utils/suite/validation.ts (hunk 9 above)
Comment (dev-pvl) [reply packages/suite/src/utils/suite/validation.ts]: moved, https://github.com/trezor/trezor-suite/pull/14713/commits/03dbbc780d1e2c8bcce13f33e84bcfa328df7e64

File: packages/suite/src/utils/suite/stake.ts (hunk 10 above)
Comment (dev-pvl) [reply packages/suite/src/utils/suite/stake.ts]: agree

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx (hunk 11 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: done, https://github.com/trezor/trezor-suite/pull/14713/commits/03dbbc780d1e2c8bcce13f33e84bcfa328df7e64

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx (hunk 12 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/UnstakeModal/UnstakeModal.tsx]: removed, https://github.com/trezor/trezor-suite/pull/14713/commits/03dbbc780d1e2c8bcce13f33e84bcfa328df7e64

**Comment by dev-pvl** [comment]
https://github.com/trezor/trezor-suite/pull/14713/commits/03dbbc780d1e2c8bcce13f33e84bcfa328df7e64 Removed the empty space under fee. Also, I added changes to the gain calculation.

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+                        </Text>
+
+                        <Paragraph align="right">
+                            <FiatValue amount={value} symbol={account.symbol} />
+                        </Paragraph>
+                    </Grid>
+                ))}
+                <Paragraph typographyStyle="body" variant="tertiary" align="center">
+                    <Translation
+                        id="TR_STAKING_YOUR_EARNINGS"
```
Comment (tomasklim) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx:84]: Please do not split 1 sentense into more translation strings 
Check e.g. string `TR_CHECKSUM_CONVERSION_INFO`

**Review by tomasklim — CHANGES_REQUESTED** [review]
LGTM, just this small change and good to good

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx (hunk 13 above)
Comment (dev-pvl) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/StakeModal/StakingInfoCards/EstimatedGains.tsx:84]: done, https://github.com/trezor/trezor-suite/pull/14713/commits/291aa34fd6ccc8c5ce123bbf40f4a7c35a18481f

**Review by tomasklim — APPROVED** [review]
squash please and let's ship it

**Pull request merged**